portal-impl/src/com/liferay/portal/model/impl/UserImpl.java

```diff
@@ -905,7 +905,12 @@ public void setDigest(String digest) {
 
 	@Override
 	public void setLanguageId(String languageId) {
-		_locale = LocaleUtil.fromLanguageId(languageId);
+		if (isDefaultUser()) {
+			_locale = LocaleUtil.fromLanguageId(languageId, false);
```

Comment:
@marianoalvarosaiz Could you point out which part in `LanguageImpl$ComapanyLocalesBag` constructor are you referring to?

Reply:
@dantewang https://github.com/liferay/liferay-portal/blob/master/portal-impl/src/com/liferay/portal/language/LanguageImpl.java#L2086-L2109

Reply:
@marianoalvarosaiz Thanks!